R/arrange.R

```diff
@@ -66,3 +71,26 @@ duckplyr_arrange <- function(.data, ...) {
   class(out) <- setdiff(class(out), "duckplyr_df")
   out
 }
+
+handle_desc <- function(dots) {
+  # Handles calls to 'desc' function by
+  # - extracting the sort order
+  # - removing any desc-function calls from the expressions: desc(colname) -> colname
+  ascending <- rep(TRUE, length(dots))
+
+  for (i in seq_along(dots)) {
+    expr <- get_expr(dots[[i]])
+
+    if (!is.call(expr))       next
+    if (expr[[1]] != "desc")  next
+
+    # Check that desc is called with a single argument
+    # (we cannot return more than one argument due to expectations of rel_translate_dots)
+    if (length(expr) > 2) cli::cli_abort("desc() must be called with exactly one argument.")
+
+    ascending[i] <- FALSE
+    dots[[i]]    <- new_quosure(expr[[2]])
```

Comment:
Do we need to use the original environment here?

Reply:
Thanks! Good point.